From akpm@linux-foundation.org  Tue Jun  2 12:01:30 2009
From: Daniel Mack <daniel@caiaq.de>
Date: Tue, 12 May 2009 13:50:34 -0700
Subject: USB: imx_udc: fix leak in imx_ep_alloc_request()
To: greg@kroah.com
Cc: akpm@linux-foundation.org, daniel@caiaq.de, augulis.darius@gmail.com, david-b@pacbell.net, eemike@gmail.com
Message-ID: <200905122109.n4CL9o83011963@imap1.linux-foundation.org>


From: Daniel Mack <daniel@caiaq.de>

cppcheck found another leak in drivers/usb/gadget/imx_udc.c

Cc: Mike Lee <eemike@gmail.com>
Cc: Darius Augulis <augulis.darius@gmail.com>
Signed-off-by: Daniel Mack <daniel@caiaq.de>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/imx_udc.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/drivers/usb/gadget/imx_udc.c
+++ b/drivers/usb/gadget/imx_udc.c
@@ -734,9 +734,12 @@ static struct usb_request *imx_ep_alloc_
 {
 	struct imx_request *req;
 
+	if (!usb_ep)
+		return NULL;
+
 	req = kzalloc(sizeof *req, gfp_flags);
-	if (!req || !usb_ep)
-		return 0;
+	if (!req)
+		return NULL;
 
 	INIT_LIST_HEAD(&req->queue);
 	req->in_use = 0;
